context/ChainsContext/service.tsx

```diff
@@ -0,0 +1,149 @@
+import { getChainFromEnvfile, getChainFromStorage, getChainFromUrl } from "./storage";
+import { ChainInfo, ChainItems } from "./types";
+
+const chainsUrl = "https://api.github.com/repos/cosmos/chain-registry/contents";
+const testnetsUrl = "https://api.github.com/repos/cosmos/chain-registry/contents/testnets";
+const registryGhUrl = "https://cdn.jsdelivr.net/gh/cosmos/chain-registry@master/";
+
+const getChains = async (chainUrl: string) => {
+  const response = await fetch(chainUrl);
+  if (!response.ok) {
+    throw new Error("Failed to get chains from registry");
+  }
+
+  const chainItems: readonly GithubChainRegistryItem[] = await response.json();
+  return chainItems;
+};
+
+export const getChainItemsFromRegistry: () => Promise<ChainItems> = async () => {
+  const [mainnets, testnets] = await Promise.all([getChains(chainsUrl), getChains(testnetsUrl)]);
+
+  const nonChainsFilter = (item: GithubChainRegistryItem) =>
+    item.type === "dir" && !item.name.startsWith(".") && !item.name.startsWith("_");
+
+  return {
+    mainnets: mainnets.filter(nonChainsFilter),
+    testnets: testnets.filter(nonChainsFilter),
+  };
+};
+
+export const getChainItemFromRegistry = async (chainName: string, isTestnet?: boolean) => {
+  const chainGhPath = isTestnet ? "testnets/" + chainName : chainName;
+  const chainGhUrl = registryGhUrl + chainGhPath + "/chain.json";
+
+  const response = await fetch(chainGhUrl);
+  if (!response.ok) {
+    throw new Error(`Failed to get ${chainName} chain from registry`);
+  }
+
+  const chain: RegistryChain = await response.json();
+  return chain;
+};
+
+export const getAssetItemsFromRegistry = async (chainName: string, isTestnet?: boolean) => {
+  const assetsGhPath = isTestnet ? "testnets/" + chainName : chainName;
+  const assetsGhUrl = registryGhUrl + assetsGhPath + "/assetlist.json";
+
+  const response = await fetch(assetsGhUrl);
+  if (!response.ok) {
+    throw new Error(`Failed to get assets for ${chainName} chain from registry`);
+  }
+
+  const assets: readonly RegistryAsset[] = (await response.json()).assets;
+  return assets;
+};
+
+const getNodeFromArray = async (nodeArray: readonly RegistryChainApisRpc[]) => {
+  // only return https connections
+  const secureNodes = nodeArray
+    .filter(({ address }) => address.startsWith("https://"))
+    .map(({ address }) => address);
+
+  if (!secureNodes.length) {
+    throw new Error("No SSL enabled RPC nodes available for this chain");
+  }
+
+  for (const node of secureNodes) {
+    try {
+      // test client connection
+      const client = await StargateClient.connect(node);
+      await client.getHeight();
+      return node;
+    } catch {}
+  }
+
+  throw new Error("No RPC nodes available for this chain");
+};
+
+const getExplorerFromArray = (explorers: readonly RegistryChainExplorer[]) => {
+  return explorers[0]?.tx_page ?? "";
+};
+
+export const getChainFromRegistry = async (chainName: string, isTestnet?: boolean) => {
```

Comment:
Ah nice and because we use the chain name we only need to query the detail info for the chain we use.